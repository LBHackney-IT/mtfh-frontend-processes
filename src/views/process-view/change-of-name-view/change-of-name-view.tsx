import React from "react";
import { Link as RouterLink } from "react-router-dom";

import { locale, processes } from "../../../services";
import { ProcessSideBarProps } from "../../../types";
import { RequestDocumentsView } from "./states/request-documents-view";

import { usePerson } from "@mtfh/common/lib/api/person/v1";
import { Process } from "@mtfh/common/lib/api/process/v1";
import {
  Button,
  Center,
  ErrorSummary,
  Spinner,
  Step,
  Stepper,
} from "@mtfh/common/lib/components";

import "./styles.scss";

const processConfig = processes.changeofname;

const { states } = processConfig;
const { enterNewName, nameSubmitted } = states;

const components = {
  [enterNewName.state]: RequestDocumentsView,
  [nameSubmitted.state]: RequestDocumentsView,
};

const { views } = locale;
const { changeofname } = views;

const getActiveStep = () => {
  return 0;
};

export const ChangeOfNameSideBar = (props: ProcessSideBarProps) => {
  const {
    process: { processId, processName },
    setCloseProcessDialogOpen,
    setCancel,
  } = props;

  const activeStep = getActiveStep();
  const steps: JSX.Element[] = [
    <Step key="step-breach-of-tenancy">{changeofname.steps.tenantsNewName}</Step>,
    <Step key="step-request-documents">{changeofname.steps.requestDocuments}</Step>,
    <Step key="step-review-documents">{changeofname.steps.reviewDocuments}</Step>,
    <Step key="step-submit-case">{changeofname.steps.submitCase}</Step>,
    <Step key="step-finish">{changeofname.steps.finish}</Step>,
  ];
  const startIndex = 0;

  return (
    <>
      <Stepper
        data-testid="mtfh-stepper-change-of-name"
        activeStep={activeStep}
        startIndex={startIndex}
      >
        {steps}
      </Stepper>
      <Button variant="secondary">{changeofname.actions.reassignCase}</Button>
      <Button
        variant="secondary"
        onClick={() => {
          setCancel(true);
          setCloseProcessDialogOpen(true);
        }}
      >
        {changeofname.actions.cancelProcess}
      </Button>
      <Button
        variant="secondary"
        as={RouterLink}
        to={`/activities/process/${processName}/${processId}`}
      >
        {changeofname.actions.caseActivityHistory}
      </Button>
    </>
  );
};

const getComponent = (process) => {
  const {
    currentState: { state },
  } = process;

  return components[state];
};

export const ChangeOfNameView = ({
  process,
  mutate,
  optional,
}: {
  process: Process;
  mutate: any;
  optional: any;
}): JSX.Element => {
  const { closeProcessReasonFinal, submitted, setSubmitted, closeCase, setCloseCase } =
    optional;
  const { error, data: person } = usePerson(process.targetId);

  if (error) {
    return (
      <ErrorSummary
        id="entity-summary"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!person) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const Component = getComponent(process);

  if (!Component) {
    return (
      <ErrorSummary
        id="change-of-name-view"
        title={locale.errors.unableToFindState}
        description={locale.errors.unableToFindStateDescription}
      />
    );
  }

  return (
    <Component
      processConfig={processConfig}
      process={process}
      mutate={mutate}
      optional={{
        person,
        submitted,
        setSubmitted,
        closeCase,
        setCloseCase,
        closeProcessReasonFinal,
      }}
    />
  );
};
