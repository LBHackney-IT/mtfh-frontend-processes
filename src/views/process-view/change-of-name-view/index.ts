export * from "./change-of-name-view";
export * from "./states";
