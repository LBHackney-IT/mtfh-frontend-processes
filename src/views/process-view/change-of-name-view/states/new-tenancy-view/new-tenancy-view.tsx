import React, { useState } from "react";

import { isPast } from "date-fns";

import {
  AppointmentDetails,
  AppointmentForm,
  CloseProcessView,
  ContactDetails,
} from "../../../../../components";
import { locale } from "../../../../../services";
import { Trigger } from "../../../../../services/processes/types";
import { IProcess, ProcessComponentProps } from "../../../../../types";
import { getPreviousState } from "../../../../../utils/processUtil";

import { Button, Heading, Text } from "@mtfh/common/lib/components";

interface NewTenancyViewProps extends ProcessComponentProps {
  processConfig: IProcess;
}

const { views } = locale;
export const NewTenancyView = ({
  processConfig,
  process,
  mutate,
  setGlobalError,
  optional,
}: NewTenancyViewProps): JSX.Element => {
  const {
    hoApprovalPassed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    nameUpdated,
    processClosed,
    processCancelled,
  } = processConfig.states;
  const { currentState } = process;
  const [needAppointment, setNeedAppointment] = useState<boolean>(
    hoApprovalPassed.state === process.currentState.state,
  );
  const [documentsSigned, setDocumentsSigned] = useState<boolean>();
  const { closeCase, setCloseCase, person, closeProcessReason } = optional;

  const processState = [processClosed.state, processCancelled.state].includes(
    currentState.state,
  )
    ? getPreviousState(process)
    : currentState;

  const formData = processState.processData.formData as {
    appointmentDateTime: string;
  };

  return (
    <div data-testid="changeofname-new-tenancy-view">
      {!closeProcessReason && (
        <>
          {!documentsSigned && (
            <Heading variant="h2">
              {views.tenureInvestigation.hoApprovedNextSteps(
                locale.views.hoReviewModal[process.processName.toLowerCase()],
              )}
            </Heading>
          )}
          {![
            tenureAppointmentScheduled.state,
            tenureAppointmentRescheduled.state,
          ].includes(process.currentState.state) && (
            <Text>{views.tenureInvestigation.mustMakeAppointment}</Text>
          )}
        </>
      )}

      {!documentsSigned &&
        [tenureAppointmentScheduled.state, tenureAppointmentRescheduled.state].includes(
          processState.state,
        ) && (
          <AppointmentDetails
            currentState={processState}
            previousStates={process.previousStates}
            needAppointment={needAppointment}
            setNeedAppointment={setNeedAppointment}
            closeCase={
              closeCase ||
              [processCancelled.state, processClosed.state].includes(currentState.state)
            }
            setCloseCase={setCloseCase}
            options={{
              requestAppointmentTrigger: Trigger.ScheduleTenureAppointment,
              rescheduleAppointmentTrigger: Trigger.RescheduleTenureAppointment,
              appointmentRequestedState: tenureAppointmentScheduled.state,
              appointmentRescheduledState: tenureAppointmentRescheduled.state,
              closeCaseButton: true,
            }}
          />
        )}

      {!documentsSigned &&
        !closeProcessReason &&
        currentState.state !== nameUpdated.state &&
        person && (
          <ContactDetails
            fullName={`${person.firstName} ${person.surname}`}
            personId={person.id}
          />
        )}

      {!closeProcessReason && (
        <AppointmentForm
          process={process}
          mutate={mutate}
          needAppointment={needAppointment}
          setGlobalError={setGlobalError}
          setNeedAppointment={setNeedAppointment}
          options={{
            buttonText: "Continue",
            requestAppointmentTrigger: Trigger.ScheduleTenureAppointment,
            rescheduleAppointmentTrigger: Trigger.RescheduleTenureAppointment,
            appointmentRequestedState: tenureAppointmentScheduled.state,
            appointmentRescheduledState: tenureAppointmentRescheduled.state,
          }}
        />
      )}

      {!documentsSigned &&
        !closeCase &&
        [tenureAppointmentScheduled.state, tenureAppointmentRescheduled.state].includes(
          process.currentState.state,
        ) &&
        !needAppointment && (
          <Button
            disabled={!isPast(new Date(formData.appointmentDateTime))}
            onClick={() => setDocumentsSigned(true)}
          >
            {views.tenureInvestigation.documentsSigned}
          </Button>
        )}

      {(documentsSigned || currentState.state === nameUpdated.state) && (
        <CloseProcessView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          statusBox={false}
          trigger={Trigger.UpdateName}
        />
      )}
    </div>
  );
};
