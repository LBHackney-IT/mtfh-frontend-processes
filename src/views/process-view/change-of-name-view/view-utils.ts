import { locale, processes } from "../../../services";

const { reviewDocuments } = locale.views;

const { states } = processes.changeofname;

const { interviewScheduled, interviewRescheduled, hoApprovalFailed, hoApprovalPassed } =
  states;

export const changeOfNameDocuments = [
  reviewDocuments.marriageCertificate,
  reviewDocuments.civilPartnershipCertificate,
  reviewDocuments.decreeAbsolute,
  reviewDocuments.finalOrder,
  reviewDocuments.deedPollDocument,
  reviewDocuments.statuoryDeclaration,
];

export const cancelButtonStates = [
  interviewScheduled.state,
  interviewRescheduled.state,
  hoApprovalFailed.state,
  hoApprovalPassed.state,
];
