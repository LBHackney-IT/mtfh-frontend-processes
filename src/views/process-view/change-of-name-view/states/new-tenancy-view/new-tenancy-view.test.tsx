import {
  getContactDetailsV2,
  getReferenceDataV1,
  mockContactDetailsV2,
  mockPersonV1,
  mockProcessV1,
  patchProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale, processes } from "../../../../../services";
import { ChangeOfNameView } from "../../change-of-name-view";
import { NewTenancyView } from "./new-tenancy-view";

let submitted = false;
let closeCase = false;
const setSubmitted = () => {};
const setCloseCase = () => {};

const options = {
  url: "/processes/changeofname/e63e68c7-84b0-3a48-b450-896e2c3d7735",
  path: "/processes/:processName/:processId",
};

const mockTenureAppointmentSchedule = (appointmentDateTime) => {
  return {
    ...mockProcessV1,
    currentState: {
      ...mockProcessV1.currentState,
      processData: {
        formData: {
          appointmentDateTime,
        },
        documents: [],
      },
      state: "TenureAppointmentScheduled",
    },
  };
};

describe("tenure-investigation-view", () => {
  beforeEach(() => {
    jest.resetModules();
    submitted = false;
    closeCase = false;
  });

  test("it renders NewTenancy view correctly for state=HOApprovalPassed", async () => {
    server.use(getReferenceDataV1({}, 200));
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <NewTenancyView
        processConfig={processes.changeofname}
        process={{
          ...mockProcessV1,
          processName: "changeofname",
          currentState: {
            ...mockProcessV1.currentState,
            state: "HOApprovalPassed",
          },
        }}
        mutate={() => {}}
        optional={{
          submitted,
          setSubmitted,
          closeCase,
          setCloseCase,
          person: mockPersonV1,
        }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    expect(container).toMatchSnapshot();
  });

  test("it renders NewTenancy view correctly for state=TenureAppointmentScheduled", async () => {
    server.use(getReferenceDataV1({}, 200));
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <NewTenancyView
        processConfig={processes.changeofname}
        process={{
          ...mockTenureAppointmentSchedule("2099-10-12T08:59:00.000Z"),
          processName: "changeofname",
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.documentsSigned, {
        exact: false,
      }),
    ).resolves.toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  test("it renders TenureInvestigation view correctly for state=TenureAppointmentScheduled, date has passed, submit fails", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(patchProcessV1({}, 500));
    render(
      <ChangeOfNameView
        process={{
          ...mockTenureAppointmentSchedule("2010-10-12T08:59:00.000Z"),
          processName: "changeofname",
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    const documentsSigned = screen.getByText(
      locale.views.tenureInvestigation.documentsSigned,
    );
    expect(documentsSigned).toBeEnabled();
    await userEvent.click(documentsSigned);
    await userEvent.click(screen.getByText(locale.views.closeProcess.outcomeLetterSent));
    await userEvent.click(screen.getByText(locale.confirm));
    await expect(
      screen.findByText("There was a problem with completing the action", {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
  });

  test("it renders TenureInvestigation view correctly for state=TenureAppointmentRescheduled", async () => {
    server.use(getReferenceDataV1({}, 200));
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <NewTenancyView
        processConfig={processes.changeofname}
        process={{
          ...mockProcessV1,
          processName: "changeofname",
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureAppointmentRescheduled",
            processData: {
              formData: {
                appointmentDateTime: "2099-10-12T08:59:00.000Z",
              },
              documents: [],
            },
          },
          previousStates: [
            {
              ...mockProcessV1.currentState,
              state: "TenureAppointmentScheduled",
              processData: {
                formData: {
                  appointmentDateTime: "2010-10-12T08:59:00.000Z",
                },
                documents: [],
              },
            },
          ],
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.documentsSigned),
    ).resolves.toBeDisabled();
    expect(container).toMatchSnapshot();
  });
});
