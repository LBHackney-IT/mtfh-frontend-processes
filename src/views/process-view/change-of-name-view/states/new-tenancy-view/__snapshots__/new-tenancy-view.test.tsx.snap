// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tenure-investigation-view it renders NewTenancy view correctly for state=HOApprovalPassed 1`] = `
<div>
  <div
    data-testid="changeofname-new-tenancy-view"
  >
    <h2
      class="lbh-heading-h2"
    >
      Change of name application approved, next steps:
    </h2>
    <p
      class="lbh-body-m"
    >
      You must make an office appointment with the applicant to sign a new tenancy agreement
    </p>
    <p
      class="lbh-body-s"
    >
      Haylee Macejkovic
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        075949281
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        Gay.Jacobi@hotmail.com
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      If the contact details are not up to date, please
       
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/processes/changeofname/e63e68c7-84b0-3a48-b450-896e2c3d7735"
      >
        update the contact details,
         
      </a>
      it will automatically update the tenantâ€™s contact details as well.
    </p>
    <div
      class="govuk-checkboxes__item"
    >
      <input
        checked=""
        class="govuk-checkboxes__input"
        id="condition"
        type="checkbox"
      />
      <label
        class="govuk-label govuk-checkboxes__label"
        for="condition"
      >
        I have made an appointment to sign the CN7 form
      </label>
    </div>
    <form
      action="#"
      class="request-appointment-form"
      id="request-appointment-form"
      novalidate=""
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="appointment-form-date-time-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="appointment-form-date-time-form-group-field"
        >
          
        </label>
        <div
          class="govuk-form-group lbh-form-group"
          id="appointment-form-date-time-field-form-group"
          style="display: flex; flex-wrap: wrap; align-items: flex-end; margin-top: 0px;"
        >
          <label
            class="govuk-label lbh-label"
            for="appointment-form-date-time-field-form-group-field"
          >
            
          </label>
          <fieldset
            class="govuk-form-group lbh-form-group mtfh-appointment-form__date"
            id="appointment-form-date"
            style="flex: 1 1 280px;"
          >
            <legend
              class="govuk-label lbh-label"
              for="appointment-form-date-field"
            >
              Date
              <sup
                aria-hidden="true"
              >
                *
              </sup>
            </legend>
            <div
              class="govuk-date-input lbh-date-input"
            >
              <div
                class="govuk-date-input__item"
              >
                <label
                  class="govuk-label lbh-label"
                  for="appointment-form-date-field-day"
                />
                <input
                  aria-label="Day"
                  aria-valuemax="31"
                  aria-valuemin="1"
                  class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                  name="day"
                  placeholder="dd"
                  required=""
                  role="spinbutton"
                  value=""
                />
              </div>
              <div
                class="govuk-date-input__item"
              >
                <label
                  class="govuk-label lbh-label"
                  for="appointment-form-date-field-month"
                />
                <input
                  aria-label="Month"
                  aria-valuemax="12"
                  aria-valuemin="1"
                  class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                  name="month"
                  placeholder="mm"
                  required=""
                  role="spinbutton"
                  value=""
                />
              </div>
              <div
                class="govuk-date-input__item"
              >
                <label
                  class="govuk-label lbh-label"
                  for="appointment-form-date-field-year"
                />
                <input
                  aria-label="Year"
                  class="govuk-input lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                  name="year"
                  placeholder="yyyy"
                  required=""
                  role="spinbutton"
                  value=""
                />
              </div>
            </div>
          </fieldset>
          <fieldset
            class="govuk-form-group lbh-form-group mtfh-appointment-form__time"
            id="appointment-form-time"
            style="flex: 1 1 280px;"
          >
            <legend
              class="govuk-label lbh-label"
              for="appointment-form-time-field"
            >
              Time
              <sup
                aria-hidden="true"
              >
                *
              </sup>
            </legend>
            <div
              class="govuk-date-input lbh-date-input"
            >
              <div
                class="govuk-date-input__item"
              >
                <label
                  class="govuk-label lbh-label"
                  for="appointment-form-time-field-hour"
                />
                <input
                  aria-label="Hour"
                  aria-valuemax="12"
                  aria-valuemin="0"
                  class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                  name="hour"
                  placeholder="00"
                  required=""
                  role="spinbutton"
                  value=""
                />
              </div>
              <div
                class="govuk-date-input__item"
              >
                <label
                  class="govuk-label lbh-label"
                  for="appointment-form-time-field-minute"
                />
                <input
                  aria-label="Minute"
                  aria-valuemax="59"
                  aria-valuemin="0"
                  class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                  name="minute"
                  placeholder="00"
                  required=""
                  role="spinbutton"
                  value=""
                />
              </div>
              <div
                class="govuk-date-input__item"
              >
                <label
                  class="govuk-label lbh-label"
                  for="appointment-form-time-field-amPm"
                />
                <select
                  aria-label="AM/PM"
                  class="govuk-select lbh-select"
                  id="amPm"
                  name="amPm"
                >
                  <option
                    value=""
                  >
                    AM/PM
                  </option>
                  <option
                    value="am"
                  >
                    AM
                  </option>
                  <option
                    value="pm"
                  >
                    PM
                  </option>
                </select>
              </div>
            </div>
          </fieldset>
        </div>
      </div>
      <button
        class="govuk-button lbh-button"
        disabled=""
        style="width: 245px;"
        type="submit"
      >
        Continue
      </button>
    </form>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders NewTenancy view correctly for state=HOApprovalPassed and closeProcessReason provided 1`] = `
<div>
  <div
    data-testid="changeofname-new-tenancy-view"
  />
</div>
`;

exports[`tenure-investigation-view it renders NewTenancy view correctly for state=TenureAppointmentScheduled 1`] = `
<div>
  <div
    data-testid="changeofname-new-tenancy-view"
  >
    <h2
      class="lbh-heading-h2"
    >
      Change of name application approved, next steps:
    </h2>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment scheduled
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
          <div
            style="display: flex; flex-direction: column; align-items: flex-start; margin-top: 15px;"
          >
            <button
              class="govuk-link lbh-link"
              type="button"
            >
              Change
            </button>
          </div>
        </div>
      </div>
    </div>
    <button
      class="govuk-button lbh-button"
      disabled=""
      type="button"
    >
      Documents signed
    </button>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders NewTenancy view correctly for state=TenureAppointmentScheduled and closeProcessReason provided 1`] = `
<div>
  <div
    data-testid="changeofname-new-tenancy-view"
  >
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment missed
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders TenureInvestigation view correctly for state=NameUpdated 1`] = `
<div>
  <div
    data-testid="changeofname-new-tenancy-view"
  >
    <h3
      class="lbh-heading-h3"
    >
      Thank you for your confirmation
    </h3>
    <ul
      class="lbh-list lbh-list--bullet"
      style="margin-left: 1em;"
    >
      <li>
        <p
          class="lbh-body-s"
        >
          This case is now closed and we have recorded this on the system - that you have sent an outcome letter to the resident. The outcome can be viewed in the activity history.
        </p>
      </li>
    </ul>
    <div
      style="margin-top: 1em;"
    >
      <a
        class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
        href="/"
      >
        Return to home page
      </a>
    </div>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders TenureInvestigation view correctly for state=ProcessClosed 1`] = `
<div>
  <h2
    class="lbh-heading-h2"
  >
    <div>
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
        target="_blank"
      >
        Haylee
         
        Macejkovic
      </a>
    </div>
  </h2>
  <div
    data-testid="changeofname-new-tenancy-view"
  >
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment missed
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Tuesday 12th October 2010
            <br />
            Time: 
            08:59 am
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mtfh-box mtfh-box--warning"
  >
    <div
      class="mtfh-status-box"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#BE3A34"
          stroke-width="1.5"
        />
        <path
          d="M24.2969 26.6587H20.6035L19.8315 11.8701H25.0688L24.2969 26.6587ZM19.771 31.8354C19.771 30.9878 19.998 30.347 20.4521 29.9131C20.9062 29.4792 21.5672 29.2622 22.4351 29.2622C23.2726 29.2622 23.9185 29.4842 24.3726 29.9282C24.8368 30.3722 25.0688 31.008 25.0688 31.8354C25.0688 32.6326 24.8368 33.2633 24.3726 33.7275C23.9084 34.1816 23.2625 34.4087 22.4351 34.4087C21.5874 34.4087 20.9315 34.1867 20.4673 33.7427C20.0031 33.2886 19.771 32.6528 19.771 31.8354Z"
          fill="#BE3A34"
        />
      </svg>
      <div>
        <div
          class="mtfh-status-heading__title"
        >
          <h4
            class="lbh-heading-h4"
          >
            Change of name application closed 
          </h4>
        </div>
      </div>
    </div>
  </div>
  <h3
    class="lbh-heading-h3"
  >
    Thank you for your confirmation
  </h3>
  <ul
    class="lbh-list lbh-list--bullet"
    style="margin-left: 1em;"
  >
    <li>
      <p
        class="lbh-body-s"
      >
        This case is now closed and we have recorded this on the system - that you have sent an outcome letter to the resident. The outcome can be viewed in the activity history.
      </p>
    </li>
  </ul>
  <div
    style="margin-top: 1em;"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="/"
    >
      Return to home page
    </a>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders TenureInvestigation view correctly for state=TenureAppointmentRescheduled 1`] = `
<div>
  <div
    data-testid="changeofname-new-tenancy-view"
  >
    <h2
      class="lbh-heading-h2"
    >
      Change of name application approved, next steps:
    </h2>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment missed
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Tuesday 12th October 2010
            <br />
            Time: 
            08:59 am
          </p>
        </div>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment scheduled
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
          <div
            style="display: flex; flex-direction: column; align-items: flex-start; margin-top: 15px;"
          >
            <button
              class="govuk-link lbh-link"
              type="button"
            >
              Change
            </button>
          </div>
        </div>
      </div>
    </div>
    <button
      class="govuk-button lbh-button"
      disabled=""
      type="button"
    >
      Documents signed
    </button>
  </div>
</div>
`;
